File: src/main/java/com/skidders/sigma/screens/clickgui/components/ButtonLineComponent.java
package com.skidders.sigma.screens.clickgui.components;

import net.minecraft.client.gui.screen.Screen;
import net.minecraft.client.util.math.MatrixStack;

import java.awt.*;

public class ButtonLineComponent {

    public final String text;
    public final float x, y;
    private final Color color;

    private final Screen parent;

    public ButtonLineComponent(String text, float x, float y, Color color, Screen parent) {
        this.text = text;
        this.x = x;
        this.y = y;
        this.color = color;
        this.parent = parent;
    }


    public void draw(MatrixStack matrices, int mouseX, int mouseY) {

    }

    public boolean click(double mouseX, double mouseY, int button) {
        return true;
    }

}

File: src/main/kotlin/com/skidders/sigma/screens/clickgui/components/ButtonLineComponent.kt
package com.skidders.sigma.screens.clickgui.components

import net.minecraft.client.gui.screen.Screen
import net.minecraft.client.util.math.MatrixStack

import java.awt.*

class ButtonLineComponent {

    val text: String
    val x: Float
    val y: Float
    private val color: Color
    private val parent: Screen

    constructor(text: String, x: Float, y: Float, color: Color, parent: Screen) {
        this.text = text
        this.x = x
        this.y = y
        this.color = color
        this.parent = parent
    }


    fun draw(matrices: MatrixStack, mouseX: Int, mouseY: Int) {

    }

    fun click(mouseX: Double, mouseY: Double, button: Int): Boolean {
        return true
    }

}
